x-pack/solutions/search/plugins/search_homepage/public/hooks/use_search_home_page_redirect.tsx

```diff
@@ -16,17 +16,23 @@ import { generateRandomIndexName } from '../utils/indices';
 import { useKibana } from './use_kibana';
 
 export const useSearchHomePageRedirect = () => {
-  const { application, http } = useKibana().services;
+  const { application } = useKibana().services;
   const indexName = useMemo(() => generateRandomIndexName(), []);
   const { data: userPrivileges } = useUserPrivilegesQuery(indexName);
   const skipGlobalEmptyState = useMemo(() => {
     return localStorage.getItem(GLOBAL_EMPTY_STATE_SKIP_KEY) === 'true';
   }, []);
-  const { data: indicesStatus } = useIndicesStatusQuery(undefined, !skipGlobalEmptyState);
+  const {
+    data: indicesStatus,
+    isLoading,
+    error,
+  } = useIndicesStatusQuery(undefined, !skipGlobalEmptyState);
 
-  const [hasDoneRedirect, setHasDoneRedirect] = useState(() => false);
-  return useEffect(() => {
-    if (hasDoneRedirect || skipGlobalEmptyState) {
+  const [redirectChecked, setRedirectChecked] = useState(() => false);
```

Comment:
`hasDoneRedirect` seemed a bit misleading given the intended behavior. The only way we know the redirect actually happened and is rendering the new page is when this hook (i.e. it's consumer) unmounts.

Thus the most we can do before it unmounts, as far as state tracking goes, is determining whether or not we've checked that a redirect should or should not happen.